pkg/sources/github/github.go

```diff
@@ -790,34 +792,39 @@ func (s *Source) scan(ctx context.Context, installationClient *github.Client, ch
 			s.setScanOptions(s.conn.Base, s.conn.Head)
 
 			repoSize := s.repoSizes.getRepo(repoURL)
-			logger.V(2).Info(fmt.Sprintf("scanning repo %d/%d", i, len(s.repos)), "repo_size_kb", repoSize)
+			logger.V(2).Info("scanning repo", "repo_size_kb", repoSize)
 
 			now := time.Now()
-			defer func(start time.Time) {
-				logger.V(2).Info(fmt.Sprintf("scanned %d/%d repos", scanned, len(s.repos)), "repo_size", repoSize, "duration_seconds", time.Since(start).Seconds())
-			}(now)
+			scanFailed := false
+			defer func(start time.Time, failed *bool) {
+				if *failed {
+					return
+				}
+				logger.V(2).Info(fmt.Sprintf("scanned %d/%d repos", scannedCount, len(s.repos)), "duration_seconds", time.Since(start).Seconds())
+			}(now, &scanFailed)
 
 			if err = s.git.ScanRepo(ctx, repo, path, s.scanOptions, sources.ChanReporter{Ch: chunksChan}); err != nil {
 				scanErrs.Add(fmt.Errorf("error scanning repo %s: %w", repoURL, err))
+				scanFailed = true
 				return nil
 			}
 
 			githubReposScanned.WithLabelValues(s.name).Inc()
 
 			if err = s.scanComments(ctx, repoURL, chunksChan); err != nil {
 				scanErrs.Add(fmt.Errorf("error scanning comments in repo %s: %w", repoURL, err))
+				scanFailed = true
 				return nil
 			}
 
-			atomic.AddUint64(&scanned, 1)
-
+			atomic.AddUint64(&scannedCount, 1)
```

Comment:
Either this needs to be moved up beside `githubReposScanned.WithLabelValues(s.name).Inc()` or it needs to be moved down here.

Reply:
I moved it here after all of the potential failures.